app/controllers/application_controller.rb

```diff
@@ -137,4 +129,12 @@ def set_locale
   def sp_session
     session.fetch(:sp, {})
   end
+
+  def create_branded_experience
+    return unless session[:sp]
+
+    @sp_logo = current_sp.logo
+    @sp_name = decorated_session.sp_name
```

Comment:
I thought about that, but that would mean creating a `sp_name` method in the ServiceProvider model, which goes against our rule of keeping business logic, and especially view-only logic out of models. What we could do is create a `ServiceProviderDecorator` that defines the logo, name, and return_to_sp_url.